### PR TITLE
[Merged by Bors] - feat: in a preconnected space, a (finite) disjoint cover of non-empty open/closed/clopen subsets contains at most one element.

### PR DESCRIPTION
**Pull request by ocfnash**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Connected.lean [hunk 1]
```diff
@@ -863,6 +863,49 @@ theorem IsClopen.eq_univ [PreconnectedSpace α] {s : Set α} (h' : IsClopen s) (
   (isClopen_iff.mp h').resolve_left h.ne_empty
 #align is_clopen.eq_univ IsClopen.eq_univ
 
+/-- In a preconnected space, any finite disjoint cover of non-empty closed subsets has at most one
```
Comment (sgouezel) [inline Mathlib/Topology/Connected.lean:866]:
```suggestion
/-- In a preconnected space, any finite disjoint cover by non-empty closed subsets has at most one
```

**Review by sgouezel — COMMENTED** [review]
Could you add the same lemma for a cover by open sets? Maybe first give a version with clopen sets, and then deduce both versions?

**Comment by ocfnash** [comment]
> Could you add the same lemma for a cover by open sets? Maybe first give a version with clopen sets, and then deduce both versions?

Good point, now done.

File: Mathlib/Topology/Connected.lean [hunk 2]
```diff
@@ -863,6 +863,89 @@ theorem IsClopen.eq_univ [PreconnectedSpace α] {s : Set α} (h' : IsClopen s) (
   (isClopen_iff.mp h').resolve_left h.ne_empty
 #align is_clopen.eq_univ IsClopen.eq_univ
 
+section disjoint_subsets
+
+variable {s : ι → Set α} (h_nonempty : ∀ i, (s i).Nonempty) (h_disj : Pairwise (Disjoint on s))
+
+/-- In a preconnected space, any disjoint family of non-empty clopen subsets has at most one
+element. -/
+lemma Subsingleton_of_Disjoint_IsClopen [PreconnectedSpace α]
```
[the remark this answers is not in the document]
Comment (ocfnash) [reply Mathlib/Topology/Connected.lean]: Sorry for being dense: you're absolutely right.

I've just pushed a commit which tries a bit harder to get things right.

**Comment by ocfnash** [comment]
bors merge

**Pull request closed without merging**